src/installLogsPrinter.js

```diff
@@ -72,15 +76,25 @@ function installLogsPrinter(on, options = {}) {
         allMessages[data.spec][data.test] = messages;
       }
 
-      logToTerminal(messages, options);
+      if (options.printLogsToConsole === "always" ||
+          ((options.printLogsToConsole === "onFail" || typeof options.printLogsToConsole === 'undefined')
+              && state !== "passed")){
+        logToTerminal(messages, options);
+      }
+
       return null;
     },
     [CONSTANTS.TASK_NAME_OUTPUT]: () => {
       outputProcessors.forEach((processor) => {
-        processor.write(allMessages);
-        logOutputTarget(processor);
+        if (options.printLogsToFile === "always" ||
```

Comment:
I am not sure if this works correctly. Here we are processing multiple test runs from one spec. But you are using the state only one of those. Please cover this in a test where you have 2 tests (it s) in a spec file and only one is failing.

Reply:
I am not sure if I know what's the exact problem, even after running a spec with one fail and one success.